open-telemetry/opentelemetry-rust
# Flatten field + fix u64 to string serde

## Changes

This PR:
- Is a continuation of what was originally surfaced https://github.com/open-telemetry/opentelemetry-rust/pull/1856.
- Flattens some `data` and `value` fields in order to be consistent with the JSON schema implemented on the OTEL Collector
  - See [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/pdata/internal/data/protogen/metrics/v1/metrics.pb.go#L469) for the struct defs and JSON tags
  - Also see [here](https://github.com/open-telemetry/opentelemetry-proto/blob/v1.3.1/examples/metrics.json) for an example resource metric in JSON
- Adds more fields to be correctly doing serde between `u64` and `String`
- Without both of these changes the `./expected/metrics.json` file **wouldn't even deserialize correctly**.

## Testing

- Run `cargo test --workspace --all-features --test grpc_build -- --nocapture` to generate grpc files. Running the first time is expected to fail, as it will create the files.
- Run `cargo test --workspace --all-features --test metrics 'test_serde' -- --nocapture` to run the new test case introduced in this PR.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

cc @lalitb
